src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1506 @@
+      if cat:CreateName().LuaString == "Technique Text" and diagrams == 0 then
+        diagrams = cat.ID
+        diagrams_cat = cat
+        use_tech = true
+        if diag then
+          print("No Harp Diagrams category found. Using Technique Text,",diagrams)
+        else
+          print("No Harp Pedals category found. Using Technique Text,",diagrams)
+        end
+      elseif string.lower(cat:CreateName().LuaString) == "harp diagrams" and diag == true then
+        print("Found Harp Diagrams category")
+        diagrams = cat.ID
+        diagrams_cat = cat
+      elseif string.lower(cat:CreateName().LuaString) == "harp pedals" and diag == false then
+        print("Found Harp Pedals category")
+        diagrams = cat.ID
+        diagrams_cat = cat
+      end -- if cat...
+    end -- for cat
+-----
+    -- find an existing diagram 
+    --
+    textexpressiondefs:LoadAll()
+    for ted in each(textexpressiondefs) do
+      if ted.CategoryID == diagrams and ted:CreateDescription().LuaString == description.LuaString then
+        print ("Diagram found at",ted.ItemNo)
+        diag_ted = ted.ItemNo
+      end -- if ted.CategoryID
+    end -- for ted...
+    --
+    -- if there is no existing diagram...', create one
+    if diag_ted == 0 then
+      local ex_ted = finale.FCTextExpressionDef()
+      local ted_text = finale.FCString()
+      if diag then
+        local text_font = diagram_font
+        ted_text.LuaString = text_font..diagram_string.LuaString
+      else 
+        ted_text.LuaString = diagram_string.LuaString
+      end -- if diag
+      ex_ted:AssignToCategory(diagrams_cat)
+      ex_ted:SetDescription(description)
+      ex_ted:SaveNewTextBlock(ted_text)
+      if use_tech then -- If using the Techniques category, override positioning
+        ex_ted:SetUseCategoryPos(false)
+        --ex_ted.HorizontalJustification = 1 -- Justify Center
+        --ex_ted.HorizontalAlignmentPoint = 5 -- Center on Music
+        --ex_ted.HorizontalOffset = nudge
+        ex_ted.VerticalAlignmentPoint = 9 -- align to 
+        ex_ted.VerticalBaselineOffset = 12
+        ex_ted.VerticalEntryOffset = -36
+      end
+      ex_ted:SaveNew()
+      diag_ted = ex_ted.ItemNo
+    end -- if diag_ted == 0
+
+-- Test to see if diagram is there already...
+    expressions:LoadAllForRegion(region)
+    for e in each(expressions) do
+      local ted = e:CreateTextExpressionDef()
+      local ted_desc = ted:CreateDescription()
+      if ted_desc:ContainsLuaString(desc_prefix.LuaString) then
+        diag_asn = true
+        goto error1
+      end 
+      end -- for e in expressions... ]]
+
+-- add the harp diagram
+
+      add_expression:SetStaff(staff_num)
+      add_expression:SetMeasurePos(measure_pos)
+      add_expression:SetID(diag_ted)
+      add_expression:SaveNewToCell(and_cell)
+
+-------
+      ::error1::
+      finenv.EndUndoBlock(true)
+      if error == true then -- Tried this as if error then, it broke the script
```

Comment:
The reason `if error` broke the script is that `error` is a pre-defined global function in Lua, and unlike everywhere else you are using `error` as a variable, in this function you never defined it as a local variable. This code will never take the error path, because error will never be `true`.

Generally speaking I would avoid naming variables `error`. But for this case, if you add a `local error = false` at the top of the function, it should fix problem with `if error`.

Reply:
renamed all instances of error to variables according to the function they are in (e.g. scale_error)